substrate/primitives/runtime/src/traits/mod.rs

```diff
@@ -1594,126 +1624,47 @@ pub trait SignedExtension:
-	/// [`SignedExtensionMetadata`].
-	fn metadata() -> Vec<SignedExtensionMetadata> {
-		sp_std::vec![SignedExtensionMetadata {
+	/// [`TransactionExtensionMetadata`].
+	fn metadata() -> Vec<TransactionExtensionMetadata> {
+		sp_std::vec![TransactionExtensionMetadata {
 			identifier: Self::IDENTIFIER,
 			ty: scale_info::meta_type::<Self>(),
 			additional_signed: scale_info::meta_type::<Self::AdditionalSigned>()
 		}]
 	}
-}
-
-/// Information about a [`SignedExtension`] for the runtime metadata.
-pub struct SignedExtensionMetadata {
-	/// The unique identifier of the [`SignedExtension`].
-	pub identifier: &'static str,
-	/// The type of the [`SignedExtension`].
-	pub ty: MetaType,
-	/// The type of the [`SignedExtension`] additional signed data for the payload.
-	pub additional_signed: MetaType,
-}
-
-#[impl_for_tuples(1, 12)]
-impl<AccountId, Call: Dispatchable> SignedExtension for Tuple {
-	for_tuples!( where #( Tuple: SignedExtension<AccountId=AccountId, Call=Call,> )* );
-	type AccountId = AccountId;
-	type Call = Call;
-	const IDENTIFIER: &'static str = "You should call `identifier()`!";
-	for_tuples!( type AdditionalSigned = ( #( Tuple::AdditionalSigned ),* ); );
-	for_tuples!( type Pre = ( #( Tuple::Pre ),* ); );
-
-	fn additional_signed(&self) -> Result<Self::AdditionalSigned, TransactionValidityError> {
-		Ok(for_tuples!( ( #( Tuple.additional_signed()? ),* ) ))
-	}
-
-	fn validate(
-		&self,
-		who: &Self::AccountId,
-		call: &Self::Call,
-		info: &DispatchInfoOf<Self::Call>,
-		len: usize,
-	) -> TransactionValidity {
-		let valid = ValidTransaction::default();
-		for_tuples!( #( let valid = valid.combine_with(Tuple.validate(who, call, info, len)?); )* );
-		Ok(valid)
-	}
-
-	fn pre_dispatch(
-		self,
-		who: &Self::AccountId,
-		call: &Self::Call,
-		info: &DispatchInfoOf<Self::Call>,
-		len: usize,
-	) -> Result<Self::Pre, TransactionValidityError> {
-		Ok(for_tuples!( ( #( Tuple.pre_dispatch(who, call, info, len)? ),* ) ))
-	}
 
+	/// Validate an unsigned transaction for the transaction queue.
+	///
+	/// This function can be called frequently by the transaction queue
+	/// to obtain transaction validity against current state.
+	/// It should perform all checks that determine a valid unsigned transaction,
+	/// and quickly eliminate ones that are stale or incorrect.
 	fn validate_unsigned(
-		call: &Self::Call,
-		info: &DispatchInfoOf<Self::Call>,
-		len: usize,
+		_call: &Self::Call,
+		_info: &DispatchInfoOf<Self::Call>,
+		_len: usize,
 	) -> TransactionValidity {
-		let valid = ValidTransaction::default();
-		for_tuples!( #( let valid = valid.combine_with(Tuple::validate_unsigned(call, info, len)?); )* );
-		Ok(valid)
+		Ok(ValidTransaction::default())
 	}
 
+	/// Do any pre-flight stuff for an unsigned transaction.
+	///
+	/// Note this function by default delegates to `validate_unsigned`, so that
+	/// all checks performed for the transaction queue are also performed during
+	/// the dispatch phase (applying the extrinsic).
```

Comment:
Yes, it's still valid. The default implementation is there because new-school transaction extensions don't need it, but this along with `TransactionExtension::validate_bare_compat` are there for backwards compatibility. Users who currently use `SignedExtension`s and haven't migrated yet will already have this function overloaded. This is then used in the `TransactionExtension::validate_bare_compat` of the `AsTransactionExtension` adapter.

Reply:
From what I understood, now `TransactionExtension::validate_bare_compat` is called before `TransactionExtension::pre_dispatch_bare_compat` during dispatch, so the effect is the same, but the function by default doesn't delegate to `validate_unsigned`. Anyway just sounded a bit confusing to me.  it's not a very important point.

Reply:
I thought the original comment was about the function but now I see it's about the docs. Those are the original docs and they still hold true for existing implementations of `SignedExtension` which are becoming deprecated. I'd keep it as is.